src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
A circular argument surely @stof. You provide constants for them. The point of a constant is a named, fixed, value. Providing constants for values, and then saying you care about the values themselves, is meaningless, if you have code - within the same class - defining constants of scalars, and then directly beneath them, an array that _doesn't use those same constants_. Which is correct? Your class internally now has two sources of truth.